src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((currentGood, nextGood) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return currentGood.localeCompare(nextGood);
+
+      case SortType.LENGTH:
+        return currentGood.length - nextGood.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((currentGood, nextGood) => {
+        return currentGood.localeCompare(nextGood);
+      });
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((currentGood, nextGood) => {
+        return currentGood.length - nextGood.length;
+      });
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortType, setType] = useState(SortType.NONE);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info',
+            { 'is-light': sortType !== SortType.ALPHABET },
+          )}
+          onClick={() => setType(SortType.ALPHABET)}
```

Comment:
move handlers to dedicated functions

Reply:
I didnt get the main idea how to do it. As I understand, I should create separate function for each hook setter and then, put it inside event handler?